printer.go

```diff
@@ -9,22 +9,35 @@ import (
 	"io"
 	"math"
 	"os"
-	"strconv"
-	"strings"
 	"time"
 
 	"github.com/gonum/floats"
 )
 
+var printerHeadings = [...]string{
+	"Iter",
+	"Runtime",
+	"FuncEvals",
+	"Func",
+	"GradEvals",
+	"|Gradient|∞",
+	"HessEvals",
+}
+
+const (
+	printerBaseTmpl = "%6v  %12v  %9v  %22v" // Base template for headings and values that are always printed.
```

Comment:
I reduced the default width for iterations to 5. I also had a look at time.Duration.String() and I increased the default width for Runtime to 16 which is enough for a fixed-width output up to 1 hour. PTAL, @btracey